state/state.go

```diff
@@ -179,6 +179,7 @@ type State struct {
 	apiQueue               APIMSGQueue
 	ackQueue               chan interfaces.IMsg
 	msgQueue               chan interfaces.IMsg
+	dataQueue              InMsgMSGQueue
```

Comment:
Do you need the fancy stuff a InMsgQueue has or would a channel have been just fine?

Reply:
channel would have worked